src/components/sidemenu-components/Drawer.tsx

```diff
@@ -0,0 +1,164 @@
+import { h } from "preact";
+import { AppBar } from "@mui/material";
+import { Box } from "@mui/material";
+import { CssBaseline } from "@mui/material";
+import { Divider } from "@mui/material";
+import { Drawer } from "@mui/material";
+import { IconButton } from "@mui/material";
+import ExploreIcon from "@mui/icons-material/Explore";
+import { List } from "@mui/material";
+import { ListItem } from "@mui/material";
+import { ListItemIcon } from "@mui/material";
+import { ListItemText } from "@mui/material";
+import Home from "@mui/icons-material/Home";
+import Notification from "@mui/icons-material/Notifications";
+import Logout from "@mui/icons-material/Logout";
+import MyProfile from "@mui/icons-material/Person";
+import { Toolbar } from "@mui/material";
+import { Typography } from "@mui/material";
+import MenuIcon from "@mui/icons-material/Menu";
+import Person from "@mui/icons-material/Person";
+import { Settings } from "@mui/icons-material";
+import { route } from 'preact-router';
+import { Link } from "preact-router";
+import { useState } from "preact/hooks";
+
+const drawerWidth = 240;
+
+type drawerProps = { path: string };
+
+interface Props {
+  /**
+   * Injected by the documentation to work in an iframe.
+   * You won't need it on your project.
+   */
+  window?: () => Window;
+  pageName: string;
+  navLink: string;
+//   children: any;
+//   path: string;
+}
+
+
+function DrawerMenu(props: any) {
+  const { window } = props;
+  const [mobileOpen, setMobileOpen] = useState(false);
+
+  const handleDrawerToggle = () => {
+    setMobileOpen(!mobileOpen);
+  };
+
+
+  const drawer = (
+    <div>
+      <Toolbar />
+      <Divider />
+      <List>
+        {["Home", "Explore", "My Profile", "Notifications", "Settings"].map(
+          (text, index) => (
+            <ListItem button key={text} onClick={() => {
+                if (text === 'Home') {route('/homepage')}
```

Comment:
These routes are going nowhere. can you create the respective routes? maybe that's why there are soo many files not being used?

Reply:
These routes are not being used because I had previously made them before we were using /app as the default path. I can add the routes so that some of the pages that are not being used are in use.